uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -574,15 +627,42 @@ define([
       vm.pagination.pageChanged();
     }
 
+    /**
+     * Performs a set of manipulations to prepare all data for calculation and
+     * then triggers the calculation itself if it can be calculated
+     *
+     * @return {Promise}
+     */
+    function attemptCalculateBalanceChange () {
+      updateOpeningBalance();
```

Comment:
why is this even done at this point? Shouldn't the opening balance being set immediately after an absence type is selected? Why do we wait until the calculation is about to be performed?

Reply:
Done.